gnovm/pkg/gnolang/static_analysis.go

```diff
@@ -0,0 +1,268 @@
+package gnolang
+
+import (
+	"errors"
+	"fmt"
+)
+
+type StaticAnalysis struct {
+	// contexts for switch and for
+	contexts []Context
+	// funcContexts for functions and lambdas
+	funcContexts []FuncContext
+
+	// here we accumulate errors
+	// from lambdas defined in the function declaration
+	lambdaErrs []error
```

Comment:
I think you're using the variable for non-lambdas as well. Perhaps a better name could be `errs`.